eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageTypes.scala

```diff
@@ -192,11 +193,15 @@ case class ChannelUpdate(signature: BinaryData,
                          chainHash: BinaryData,
                          shortChannelId: ShortChannelId,
                          timestamp: Long,
-                         flags: BinaryData,
+                         messageFlags: Byte,
+                         channelFlags: Byte,
                          cltvExpiryDelta: Int,
                          htlcMinimumMsat: Long,
                          feeBaseMsat: Long,
-                         feeProportionalMillionths: Long) extends RoutingMessage with HasTimestamp with HasChainHash
+                         feeProportionalMillionths: Long,
+                         htlcMaximumMsat: Option[Long]) extends RoutingMessage with HasTimestamp with HasChainHash {
+  if ((messageFlags & 1) != 0) require(htlcMaximumMsat.isDefined, "htlcMaximumMsat must be defined when messageFlags bit is set")
```

Comment:
Should we also throw an exception if bit `option_channel_htlc_max` is not defined but htlcMaximumMsat ? Our check would become:
```scala
   require(((messageFlags & 1) != 0) == htlcMaximumMsat.isDefined, "htlcMaximumMsat is not consistent with messageFlags") 
```